protoc-gen-openapiv2/internal/genopenapi/format.go

```diff
@@ -0,0 +1,43 @@
+package genopenapi
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+
+	"gopkg.in/yaml.v2"
+)
+
+type Format string
+
+const (
+	FormatJSON Format = "json"
+	FormatYAML Format = "yaml"
+)
+
+type ContentEncoder interface {
+	Encode(v interface{}) (err error)
+}
+
+func (f Format) Validate() error {
+	switch f {
+	case FormatJSON, FormatYAML:
+		return nil
+	default:
+		return errors.New("unknown format: " + string(f))
+	}
+}
+
+func (f Format) NewEncoder(w io.Writer) ContentEncoder {
+	switch f {
+	case FormatYAML:
+		return yaml.NewEncoder(w)
+	case FormatJSON:
+		fallthrough
+	default:
+		enc := json.NewEncoder(w)
+		enc.SetIndent("", "  ")
+
+		return enc
+	}
```

Comment:
It doesn't seem to me like any code paths today should be setting this to anything other than JSON or YAML, so do we need the default case?

Reply:
Yeah, we don't need it. I just was a little bit lazy to handle an error. Fixed.